File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.js
function solution(nums) {
    
    var answer = 0;
    
    // 빈 배열 선언
    var arr = [];
    
    // nums를 순회하면서 지금까지 나오지 않았던 숫자가 나오면 answer에 +1을 해줌
    for (var i = 0; i < nums.length; i++) {
        
        // 만약 nums[i]가 arr에 없다면
        if (!arr.includes(nums[i])) {
           
            // arr에 해당 값을 넣어줌
            arr.push(nums[i]);
            
            // answer에 +1 더해줌
            answer++;
            
        }
    }
    
    if(answer > nums.length / 2) {
        return nums.length / 2;
    }
    else {
        return answer;
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    n_2 = len(nums) / 2
    
    nums_set = set(nums)
    nums_set_length = len(nums_set)
    
    if nums_set_length >= n_2:
        return n_2
    else:
        return nums_set_length
    
    